src/frontend/src/pages/AdminToolsPage/TeamConfig/TeamTypeFormModal.tsx

```diff
@@ -0,0 +1,162 @@
+import { useForm } from 'react-hook-form';
+import NERFormModal from '../../../components/NERFormModal';
+import { FormControl, FormLabel, FormHelperText, Tooltip, Typography, Button } from '@mui/material';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+import { useToast } from '../../../hooks/toasts.hooks';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Box } from '@mui/system';
+import HelpIcon from '@mui/icons-material/Help';
+import { TeamType } from 'shared';
+import { CreateTeamTypePayload } from '../../../hooks/team-types.hooks';
+import React, { useState } from 'react';
+import FileUploadIcon from '@mui/icons-material/FileUpload';
+import useFormPersist from 'react-hook-form-persist';
+import { FormStorageKey } from '../../../utils/form';
+
+interface TeamTypeFormModalProps {
+  open: boolean;
+  handleClose: () => void;
+  defaulValues?: TeamType;
+  onSubmit: (data: CreateTeamTypePayload) => Promise<TeamType>;
+}
+
+const schema = yup.object().shape({
+  name: yup.string().required('Material Type is Required'),
+  iconName: yup.string().required('Icon Name is Required'),
+  description: yup.string().required('Description is Required')
+});
```

Comment:
Try out using yup.mixed() to validate the image

Reply:
whats the point of that, like if its optional anyways why do we need a validator especially if its mixed because that would allow any value to be passed in right?

Reply:
Poor wording on my part, you can use yup.mixed() as a way to validate file uploads - I mean just to validate that the file is an image